### PR TITLE
Ornl additions

### PR DESCRIPTION
**Pull request by ddahlbom**
This PR contains some functionality that will be useful to have for the ORNL workshop. In particular:

1. Sakib's form factor function is tied into the structure factor framework. Keywords `ff_elem=nothing` and `lande=false` have been added to `dynamic_structure_factor`. If a valid element is given to `ff_elem`, the form factor will be applied to the result. Additionally, there is a public function `apply_form_factor` which can be used on an already existing structure factor.
3. Added a simple function to create a new larger system from an existing smaller system. The spin configuration of the smaller system is simply extended periodically. 
4. Added a function for generating slices from a structure factor by specifying a sequence of points in momentum space. It doesn't interpolate by default, but it can do so. This adds `Interpolations.jl` as a dependency, which is quite lightweight.

Additionally,
1. The memory layout of the omega axis has been changed. It now goes from $-\omega_{\textrm{max}}$ to $\omega_{\textrm{max}}$. 0 energy corresponds to index 0.
2. Some bugs have been fixed. (Indexing problem in structure factor calculation when both `reduce_basis` and `dipole_factor` are set to `false`. Missing brackets in `brav_lattice` function.)

### CODE REVIEW

**Comment by ddahlbom** [comment]
Also, I made some minor revisions to the tests, in particular I added an explicit RNG with seed to tests where this matters.

I briefly looked into speeding up the tests. This will take more investigation. Consider the Langevin tests: If I pull out the individual tests and run them as regular scripts, each of the three tests takes significantly less than a second. However, when I call `Pkg.test("Sunny", test_args=["test_langevin"])`, the tests takes 30 seconds. If I run the entire test suite, the Langevin tests take about 7 seconds, presumably because some relevant compilation has already occurred. Perhaps this comes down to the use of StaticArrays for SU(N) when N<6, but I suspect something else may be going on.

**Review by kbarros — APPROVED** [review]
One thing that would be nice is some (at least cursory) test for the structure factor functions. Could be something relatively trivial.

File: Project.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ CrystalInfoFramework = "6007d9b0-c6b2-11e8-0510-1d10e825f3f1"
 FFTW = "7a1cc6ca-52ef-59f5-83cd-3a7055c09341"
 FilePathsBase = "48062228-2e41-5def-b9a4-89aafe57970f"
 Inflate = "d25df0c9-e2be-5dd7-82c8-3ad0b3e990b9"
+Interpolations = "a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"
```
[the remark this answers is not in the document]
Comment (ddahlbom) [reply Project.toml:12]: Yeah, I guess you're right. It's not huge, but it's not small. On my computer it took 41.9 ms (19% complication time). So, much less than DataFrames (1912 ms), but many other packages take less than 1 ms.

File: src/StructureFactors.jl [hunk 2]
```diff
@@ -227,68 +237,78 @@ function apply_dipole_factor(sf::StructureFactor)
...
     thermalize::Int=10, bz_size=(1,1,1), reduce_basis::Bool=true,
-    dipole_factor::Bool=false, dynΔt::Float64=0.01, dyn_meas::Int=100,
-    meas_rate::Int=10, verbose::Bool=false
+    dipole_factor::Bool=false, Δt::Float64=0.01, num_ωs::Int=100,
+    ff_elem=nothing, lande=false,
+    ω_max=nothing, verbose::Bool=false
 ) where {S <: AbstractSampler}
+
+    # Check that form factor element is valid before doing calculations
+    !isnothing(ff_elem) && form_factor([π,0,0], ff_elem, lande)
```
[the remark this answers is not in the document]
Comment (ddahlbom) [reply src/StructureFactors.jl:307]: I will revise.

File: src/Integrators.jl [hunk 3]
```diff
@@ -150,7 +155,7 @@ function SchrodingerMidpoint(sys::SpinSystem{N}) where N
 end
 
 function SchrodingerMidpoint(sys::SpinSystem{0})
-    @warn "SchrodingerMidpoint integration is only available for SU(N) systems. Reverting to SphericalMidpoint integrator."
+    @warn "SchrodingerMidpoint integration is only available for SU(N) systems. Using SphericalMidpoint integrator."
```
[the remark this answers is not in the document]
Comment (ddahlbom) [reply src/Integrators.jl:158]: Agree this is a good idea, agree also that I won't do it this round. I'll try to get this and maybe some elementary structure factor test in soon, maybe this weekend, but I have to switch gears a bit over the next day or so.

(Worth noting that the user who simply creates a model and uses `dynamic_structure_factor` or `langevin_sampler` will never construct an integrator directly. But fully agree with the broader point.)

File: Project.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddahlbom) [reply Project.toml:12]: Will leave it for now. As a percentage of the total, it's still quite small. Going forward, we will likely make changes to the slicing function and can remove the dependency if desired.

**Pull request merged**